packages/headless/src/features/folding/folding-utils.ts

```diff
@@ -9,11 +9,18 @@ export function getAllIncludedResultsFrom(relevantResult: ResultWithFolding) {
     .map((result) => result.parentResult!);
 
   const resultsInCollection = removeDuplicates(
-    [relevantResult, ...foldedResults, ...parentResults],
+    [...foldedResults, ...parentResults],
```

Comment:
This is kind of like calling `removingDuplicates` two times in a row, but just by enforcing order.

I'm fairly certain this would have also worked (without the new code for `includedResults`):

```typescript
const resultsInCollection = removeDuplicates(
    [...foldedResults, ...parentResults, relevantResult],
    (result) => result.uniqueId
)
```

So either we can just change the order here (and rely on the new UT to make sure the implementation of `removeDuplicates` stays with a `reduce(Left)`, or we could change `removeDuplicates` to support a parameter to control the order of the array we want prioritized (reduce left or right)